test/tests/functional/pbs_multi_sched.py

```diff
@@ -647,6 +647,268 @@ def test_multi_fairshare(self):
         self.assertEquals(n.nshares, sc3_shares)
         self.assertEquals(n.usage, sc3_usage)
 
+    @timeout(300)
+    def test_fairshare_usage(self):
+        """
+        Test the schedulers fairshare usage file and
+        check the usage file is updating correctly or not
+        """
+        self.common_setup()
+        rv = self.server.add_resource('foo', 'float', 'nh')
+        self.assertTrue(rv)
+        self.server.manager(MGR_CMD_SET, SERVER,
+                            {'job_history_enable': 'True'}, expect=True)
+        # Add entry to the resource group of default scheduler
+        self.scheds['default'].add_to_resource_group('grp1', 100, 'root', 60)
+        self.scheds['default'].add_to_resource_group('grp2', 200, 'root', 40)
+        self.scheds['default'].add_to_resource_group('pbsuser1',
+                                                     101, 'grp1', 40)
+        self.scheds['default'].add_to_resource_group('pbsuser2',
+                                                     102, 'grp1', 20)
+        self.scheds['default'].add_to_resource_group('pbsuser3',
+                                                     201, 'grp2', 30)
+        self.scheds['default'].add_to_resource_group('pbsuser4',
+                                                     202, 'grp2', 10)
+        # Add entry to the resource group of multisched 'sc1'
+        self.scheds['sc1'].add_to_resource_group('grp1', 100, 'root', 60)
+        self.scheds['sc1'].add_to_resource_group('grp2', 200, 'root', 40)
+        self.scheds['sc1'].add_to_resource_group('pbsuser1',
+                                                 101, 'grp1', 40)
+        self.scheds['sc1'].add_to_resource_group('pbsuser2',
+                                                 102, 'grp1', 20)
+        self.scheds['sc1'].add_to_resource_group('pbsuser3',
+                                                 201, 'grp2', 30)
+        self.scheds['sc1'].add_to_resource_group('pbsuser4',
+                                                 202, 'grp2', 10)
+        # Add entry to the resource group of multisched 'sc2'
+        #(sched_priv_sc2 at non-default location)
+        self.scheds['sc2'].add_to_resource_group('grp1', 100, 'root', 60)
```

Comment:
I think testing for non-default location of multisched priv dir exclusively for fairshare scenario is not required if similar kind of test exists that checks the multiched functionality where priv dir is not at default location.
If there is any test case that verifies functionality when the changes made to multisched existing in non default location. Then this verification of ['sc2'] is not required.

Reply:
Yes. we are checking functionality for non-default location of multisched in most of the test cases but there is no testcase in which checking the read/update operations on the fairshare usage file. Would you still recommend to remove the code ?

Reply:
Yeah remove it. The code will always refer to PBS_HOME for referring the location of usage file. So similar tests are already covered as you said for checking the functionality for non-default location of multisched where they look for non-default PBS_HOME.
No need to specifically check for fairshare functinality for non-default PBS_HOME